### PR TITLE
Replace TagPanel react-dnd with react-beautiful-dnd

### PR DESCRIPTION
**Pull request by lukebarnard1**
This new library handles the simple case of an ordered vertical
(or horizontal) list of items that can be reordered.

It provides animations, handles positioning of items mid-drag
and exposes a much simpler API to react-dnd (with a slight loss
of potential function, but we don't need this flexibility here
anyway).

Apart from this, TagOrderStore had to be changed in a highly
coupled way, but arguably for the better. Instead of being
updated incrementally every time an item is dragged over
another and having a separate "commit" action, the
asyncronous action `moveTag` is used to reposition the tag in
the list and both dispatch an optimistic update and carry out
the request as before. (The MatrixActions.accountData is still
used to indicate a successful reordering of tags).

The view is updated instantly, in an animated way, and this
is handled at the layer "above" React by the DND library.

### CODE REVIEW

**Comment by lukebarnard1** [comment]
And for once Travis CI is being annoying and isn't starting chrome

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,9 @@ dist: trusty
 
 # we don't need sudo, so can run in a container, which makes startup much
 # quicker.
-sudo: false
+#
+# unfortunately we do require sudo as per https://github.com/travis-ci/travis-ci/issues/8836
```
Comment (dbkr) [inline .travis.yml]: temporarily, hopefully?

**Comment by lukebarnard1** [comment]
This fixes https://github.com/vector-im/riot-web/issues/5910

**Comment by dbkr** [comment]
otherwise lgtm

File: .travis.yml (hunk 1 above)
Comment (lukebarnard1) [reply .travis.yml]: At some point they should fix the issue, sure.

**Comment by lukebarnard1** [comment]
Having tested this locally again, it seems clicking TagPanel doesn't deselect all tags. Will fix here.

**Comment by lukebarnard1** [comment]
>please comment hacky workaround like this

yep, sure, sorry totally blanked on that one.

**Pull request merged**